### PR TITLE
Fixing empty anchors

### PR DESCRIPTION
**Pull request by dav009**
Fixes Issue https://github.com/diegoceccarelli/json-wikipedia/issues/5

- When descriptions are empty it uses the wikipedia Id replacing underscores by spaces.

### CODE REVIEW

File: src/main/java/it/cnr/isti/hpc/wikipedia/article/Link.java [hunk 1]
```diff
@@ -51,16 +51,18 @@ public void setId(String id) {
...
-			return id;
+		return this.description;
 	}
 	
 	
 	public void setDescription(String description) {
-		this.description = description;
+        // Some links do not have any anchor
+        // For those cases the anchor is the same wikipedia Id
+		if (description==""){
```
Comment (diegoceccarelli) [inline src/main/java/it/cnr/isti/hpc/wikipedia/article/Link.java]: could you please use description.isEmpty() ?

File: src/test/java/it/cnr/isti/hpc/wikipedia/article/en/ArticleTest.java [hunk 2]
```diff
@@ -82,6 +83,26 @@ public void testNotRedirect() throws IOException {
...
+        parser.parse(a, mediawiki);
+
+        // No anchor should be empty
+        for (Link link:a.getLinks()){
+            assert(link.getDescription()!="");
+        }
+
+        // testing an specific anchor
+        for (Link link:a.getLinks()){
+            if (link.getId()=="HTMS_Chakri_Naruebet")
```
Comment (diegoceccarelli) [inline src/test/java/it/cnr/isti/hpc/wikipedia/article/en/ArticleTest.java]: please use .equals() for string comparison

File: src/test/java/it/cnr/isti/hpc/wikipedia/article/en/ArticleTest.java [hunk 3]
```diff
@@ -82,6 +83,26 @@ public void testNotRedirect() throws IOException {
...
+              assert(link.getDescription()=="HTMS Chakri Naruebet");
```
Comment (diegoceccarelli) [inline src/test/java/it/cnr/isti/hpc/wikipedia/article/en/ArticleTest.java]: assertEquals(expected result, result)

File: src/test/java/it/cnr/isti/hpc/wikipedia/article/en/ArticleTest.java [hunk 4]
```diff
@@ -82,6 +83,26 @@ public void testNotRedirect() throws IOException {
...
+
+    @Test
+    public void testNoEmptyAnchors() throws IOException {
+        Article a = new Article();
+        String mediawiki = IOUtils.getFileAsUTF8String("./src/test/resources/en/Royal_Thai_Armed_Forces.txt");
+        parser.parse(a, mediawiki);
+
+        // No anchor should be empty
+        for (Link link:a.getLinks()){
+            assert(link.getDescription()!="");
```
Comment (diegoceccarelli) [inline src/test/java/it/cnr/isti/hpc/wikipedia/article/en/ArticleTest.java]: assertFalse(...isEmpty())

**Comment by dav009** [comment]
;[ ahg forgot those string comparisons in java are meant to be done with `equals`.:+1: 
Fixed. Re-review

**Comment by diegoceccarelli** [comment]
thanks @dav009,
Also a minor comment, this.id.replace("_", " "), in this case you can use the char this.id.replace('_', ' '), it would be better for the memory usage ;) 
Thank you!

**Pull request merged**